lib/evm/eei.ts

```diff
@@ -46,9 +60,9 @@ export default class EEI {
    * @throws if out of gas
    */
   useGas (amount: BN): void {
-    this._runState.gasLeft.isub(amount)
-    if (this._runState.gasLeft.ltn(0)) {
-      this._runState.gasLeft = new BN(0)
+    this._gasLeft.isub(amount)
+    if (this._gasLeft.ltn(0)) {
+      this._gasLeft = new BN(0)
       trap(ERROR.OUT_OF_GAS)
     }
   }
```

Comment:
To be honest I'm not sure what you have in mind to put in the `gas.ts` module :) The `useGas` and `refundGas` methods (maybe like [this](https://github.com/ethereum/py-evm/blob/master/eth/vm/gas_meter.py))?

>  this is also used only in very isolated places and not throughout the whole class.

It's mostly used in `opFns` (and also used for instruction base fee in `Loop`). On purpose I avoided doing gas metering in `EEI`.

Reply:
Thanks for digging this out from Py-EVM, yes I was actually thinking about something like this. We can do this later though if agreed upon, don't let this distract you here. I would assume this would be useful if e.g. extended with events which could be used to real-time display gas usage in IDEs e.g..

Reply:
Great, I think this is doable. One possibility (for next versions) is to take inspiration from go-ethereum's [gas table](https://github.com/ethereum/go-ethereum/blob/master/core/vm/gas_table.go) bring most gas calculation logic there (possibly even hardfork-specific gas costs).